zebra-chain/src/block.rs

```diff
@@ -145,6 +145,8 @@ impl ZcashSerialize for BlockHeader {
         self.previous_block_hash.zcash_serialize(&mut writer)?;
         writer.write_all(&self.merkle_root_hash.0[..])?;
         writer.write_all(&self.final_sapling_root_hash.0[..])?;
+        // this is a truncating cast, rather than a saturating cast
+        // but u32 times are valid until 2106
         writer.write_u32::<LittleEndian>(self.time.timestamp() as u32)?;
```

Comment:
I don't think truncation is possible until we start generating our own timestamps. Even then, our block validation code should prevent any truncation, because the time won't be in the valid range (within 90 minutes of the previous blocks' median, and less 2 hours into the future according to the node's clock).

I'm hesitant about panics in general - if they can be triggered remotely, they can lead to a denial of service attack.

Reply:
I update the comment to talk about the block time checks in 153a96848a612b74240e7c511e3edf199dfeae7a.